Fix some linter issues

```
2020-09-30 01:04:27 [INFO  ]   File:[config/capture_lcov.sh]
2020-09-30 01:04:27 [ERROR ]   Found errors in [shellcheck] linter!
2020-09-30 01:04:27 [ERROR ]   [
In config/capture_lcov.sh line 23:
    echo "==> $1"; if test ${dry_run:-no} = "no"; then eval $1; fi
                           ^------------^ SC2086: Double quote to prevent globbing and word splitting.
                                                            ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    echo "==> $1"; if test "${dry_run:-no}" = "no"; then eval "$1"; fi


In config/capture_lcov.sh line 52:
if ! [[ -x $GCOV ]]; then GCOV=`which gcov` || die "gcov = ${GCOV} was not found"; fi
                               ^----------^ SC2006: Use $(...) notation instead of legacy backticked `...`.

Did you mean: 
if ! [[ -x $GCOV ]]; then GCOV=$(which gcov) || die "gcov = ${GCOV} was not found"; fi


In config/capture_lcov.sh line 53:
if ! [[ -x $LCOV ]]; then LCOV=`which lcov` || die "lcov = ${LCOV} was not found"; fi
                               ^----------^ SC2006: Use $(...) notation instead of legacy backticked `...`.

Did you mean: 
if ! [[ -x $LCOV ]]; then LCOV=$(which lcov) || die "lcov = ${LCOV} was not found"; fi


In config/capture_lcov.sh line 69:
$LCOV --gcov-tool $GCOV --quiet --list coverage.info &> ${repname}
                  ^---^ SC2086: Double quote to prevent globbing and word splitting.
                                                        ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
$LCOV --gcov-tool "$GCOV" --quiet --list coverage.info &> "${repname}"


In config/capture_lcov.sh line 71:
cat ${repname}
    ^--------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
cat "${repname}"

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
  https://www.shellcheck.net/wiki/SC2006 -- Use $(...) notation instead of le...]
2020-09-30 01:04:27 [ERROR ]   Linter CMD:[shellcheck --color --external-sources config/capture_lcov.sh]
```

--- a/.github/workflows/linter.yml
+++ b/.github/workflows/linter.yml
@@ -59,7 +59,6 @@ jobs:
           VALIDATE_ENV: true
           VALIDATE_HTML: true
           VALIDATE_JSON: true
-          VALIDATE_JSON: true
           VALIDATE_LATEX: true
           VALIDATE_LUA: true
           VALIDATE_PERL: true

--- a/config/capture_lcov.sh
+++ b/config/capture_lcov.sh
@@ -1,16 +1,15 @@
 #!/bin/bash
-##-*- Mode: bash -*-
-##---------------------------------------------------------------------------##
-## File  : config/capture_lcov.sh
-## Date  : Thursday, Apr 09, 2020, 10:12 am
-## Author: Kelly Thompson
-## Note  : Copyright (C) 2020, Triad National Security, LLC.
-##         All rights are reserved.
-##
-## Helper script used by the make system's 'covrep' target.  This script allows
-## us to capture the output from lcov to a file in the build directory since
-## lcov only knows how to echo its report to stdout.
-##----------------------------------------------------------------------------##
+#-*- Mode: bash -*-
+#--------------------------------------------------------------------------------------------------#
+# File  : config/capture_lcov.sh
+# Date  : Thursday, Apr 09, 2020, 10:12 am
+# Author: Kelly Thompson
+# Note  : Copyright (C) 2020, Triad National Security, LLC., All rights are reserved.
+#
+# Helper script used by the make system's 'covrep' target.  This script allows us to capture the
+# output from lcov to a file in the build directory since lcov only knows how to echo its report to
+# stdout.
+#--------------------------------------------------------------------------------------------------#
 
 # Helper functions
 
@@ -20,7 +19,7 @@ function die ()
 }
 function run ()
 {
-    echo "==> $1"; if test ${dry_run:-no} = "no"; then eval $1; fi
+    echo "==> $1"; if [[ "${dry_run:-no}" == "no" ]]; then eval "$1"; fi
 }
 function print_use()
 {
@@ -49,8 +48,8 @@ esac
 done
 
 # defaults if GCOV or LCOV are not set.
-if ! [[ -x $GCOV ]]; then GCOV=`which gcov` || die "gcov = ${GCOV} was not found"; fi
-if ! [[ -x $LCOV ]]; then LCOV=`which lcov` || die "lcov = ${LCOV} was not found"; fi
+if ! [[ -x $GCOV ]]; then GCOV=$(which gcov) || die "gcov = ${GCOV} was not found"; fi
+if ! [[ -x $LCOV ]]; then LCOV=$(which lcov) || die "lcov = ${LCOV} was not found"; fi
 
 # Sanity Checks
 
@@ -63,12 +62,11 @@ if ! [[ -f coverage.info ]]; then
 fi
 
 # Ask lcov to generate the text report:
-# - generate the text report at the same location as coverage.info (top level of
-#   build tree).
+# - generate the text report at the same location as coverage.info (top level of build tree).
 echo "$LCOV --gcov-tool $GCOV --quiet --list coverage.info --> ${repname}"
-$LCOV --gcov-tool $GCOV --quiet --list coverage.info &> ${repname}
+${LCOV} --gcov-tool "${GCOV}" --quiet --list coverage.info &> "${repname}"
 # also print the report to stdout.
-cat ${repname}
+cat "${repname}"
 
 #--------------------------------------------------------------------------------------------------#
 # end config/capture_lcov.sh

